source/dlangbot/bugzilla.d

```diff
@@ -79,3 +81,43 @@ Issue[] getDescriptions(R)(R issueRefs)
         .sort!((a, b) => a.id < b.id)
         .release;
 }
+
+shared string bugzillaLogin, bugzillaPassword;
+
+Json apiCall(string method, Json[string] params)
+{
+    import std.net.curl;
+    import vibe.stream.operations : readAllUTF8;
+    import dlangbot.utils : request;
+
+    string[string] urlParams = [
+        "method" : method,
+    ];
+    static if (!is(P == typeof(null)))
+        urlParams["params"] = params.Json.toString();
+    auto url = bugzillaURL ~ "/jsonrpc.cgi?" ~ urlEncode(urlParams);
+    auto jsonText = url.request.bodyReader.readAllUTF8;
+    auto reply = jsonText.parseJsonString();
+    enforce("error" !in reply, "Server error: " ~ reply["error"].to!string);
+    return reply["result"];
+}
+
+Json authenticatedApiCall(string method, Json[string] params)
+{
+    params["Bugzilla_login"] = bugzillaLogin;
+    params["Bugzilla_password"] = bugzillaPassword;
+    return apiCall(method, params);
+}
+
+/// Close these bug IDs as FIXED and leave a comment.
```

Comment:
Nit: order higher-level / public functions before lower level / private ones - `closeIssues` first, and then `authenticatedApiCall`, followed by `apiCall`.